### DISCUSSION TITLE
How to localize ui-data-list-wrapper

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [taiga-family/taiga-ui#9298] [Q&A] [answered]
There is an text "Nothing Found". Any idea how to localize it?
I am using custom locale (Lithuaniain) nad feeding at least grid like this:

`
  gridOptions = {
        // domLayout: 'autoHeight',
        localeText: AG_GRID_LOCALE_LT, // TODO: on language change change locale from https://www.ag-grid.com/javascript-data-grid/localisation/ or custom so it is better to do full page refresh on language change
    } as any;
`

it is maybe general idea how to localize all components with not supported locale. I have all translations in local file with values taken from Taiga-Ui github

Thanks

**Comment by waterplea** [comment, marked as answer]
Would be great if you contributed a Lithuanian translation, there's not that many texts in Taiga UI, here's English translation with all the prompts:
https://github.com/taiga-family/taiga-ui/tree/main/projects/i18n/languages/english
To provide your custom translations you need to provide `Observable<TuiLanguage>` with `TUI_LANGUAGE` token.

**Reply by vytautas-pranskunas-** [reply]
> Would be great if you contributed a Lithuanian translation, there's not that many texts in Taiga UI, here's English translation with all the prompts: https://github.com/taiga-family/taiga-ui/tree/main/projects/i18n/languages/english To provide your custom translations you need to provide `Observable<TuiLanguage>` with `TUI_LANGUAGE` token.

an contribute. Just not sure exactly how.. do a fork and PR?

**Reply by waterplea** [reply]
Yes, make a fork, copy `english` folder, name it properly and change all the prompts. Add import here:
https://github.com/taiga-family/taiga-ui/blob/main/projects/i18n/languages/index.ts
That should do it 👍

**Reply by vytautas-pranskunas-** [reply]
have added PR:

https://github.com/taiga-family/taiga-ui/pull/9307

did not check 100% of translations just went through with my eyes. If there is more accurate translation I will fix it with another PR.
When can i expect new version with language?

p.s. o noticed that i made localisation for AG_GRID not for Taiga... those late working hours... :D

**Reply by waterplea** [reply]
We typically release every Monday

**Reply by vytautas-pranskunas-** [reply]
is that version with lithuanian localization released?
could you please tell me how exaclty provide ```Observable<TuiLanguage> with TUI_LANGUAGE token.```

**Reply by waterplea** [reply]
It should be there in the latest one. Here's the docs on i18n:
https://taiga-ui.dev/i18n
